src/test/kotlin/com/atlassian/performance/tools/lib/WeightedActionMetric.kt

```diff
@@ -0,0 +1,8 @@
+package com.atlassian.performance.tools.lib
+
+import com.atlassian.performance.tools.jiraactions.api.ActionMetric
+
+/**
+ * Wrapper for ActionMetric which can store the apdex 'weight' applied alongside this metric
```

Comment:
This can be easily confused with [synthetic Apdex weighing](https://hello.atlassian.net/wiki/spaces/DC/pages/353104296/Hardware+recommendations+-+Confluence+Large+profile?focusedCommentId=362259099#comment-362259099) we used for Confluence.
`Apdex` has a `score` method, perhaps this could be called `ScoredActionMetric` or `ApdexActionMetric`?

PS. In the partial (paused and not merged) bottleneck analysis I also needed the Apdex score classification and it looks like this: 1035cf16620d4e3a7d220e01f103e638864ac724
Feel free to reuse or ignore

Reply:
I'll rename to avoid confusion.

re: https://github.com/atlassian/jira-hardware-exploration/commit/1035cf16620d4e3a7d220e01f103e638864ac724 I think there is overlap but i think they are complimentary rather than equivalent, but I'll take a look.

Reply:
Having reviewed my code further I do think there is overlap with https://github.com/atlassian/jira-hardware-exploration/commit/1035cf16620d4e3a7d220e01f103e638864ac724

i.e. later in my code I want counts of Satisfactory, Frustrated etc metrics and effectively run the filters you code into  Apdex.kt

However I want to keep the individual scores for additional calculations so I think both add useful functionality to Apdex.kt